Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -589,17 +593,74 @@ void DragCaretController::clearCaretPositionWithoutUpdatingStyle()
     clearCaretRect();
 }
 
+static void setNodeContainsSelecionEndPoint(const Position& position, bool value)
+{
+    // We use anchorNode instead of containerNode() because nodeWillBeRemoved must update position when anchored node is removed.
+    for (Node* currentNode = position.anchorNode(); currentNode; currentNode = currentNode->parentOrShadowHostNode()) {
```

Comment:
Should we use `RefPtr` for `currentNode` here?

Reply:
Nah, we don't need to use RefPtr. These functions are all trivial.